php/src/Google/Protobuf/Internal/Message.php

```diff
@@ -1597,10 +1598,19 @@ private function existField($field)
 
         $values = $this->$getter();
         if ($field->isMap()) {
+            if (($options & \Google\Protobuf\PrintOptions::EMIT_DEFAULTS)) {
+                return true;
```

Comment:
Pull this out instead of duplicating this in each of the if / elseif / else cases.